mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -0,0 +1,218 @@
+            )
+
+    variable_string = (
+        ""
+        if variables_dict is None
+        else "\n".join(
+            f"Provided {variable}: {variable_value}"
+            for variable, variable_value in variables_dict.items()
+        )
+    )
+
+    return variable_string
+
+
+def make_genai_metric(
+    name: str,
+    definition: str,
+    grading_prompt: str,
+    examples: Optional[List[EvaluationExample]] = None,
+    version: Optional[str] = "v1",
+    model: Optional[str] = "openai:/gpt4",
+    variables: Optional[List[str]] = None,
+    parameters: Optional[Dict[str, Any]] = None,
+    aggregations: Optional[List[str]] = None,
+    greater_is_better: bool = True,
+) -> EvaluationMetric:
+    """
+    Create a genai metric used to evaluate LLM using LLM as a judge in MLflow.
+
+    :param name: Name of the metric.
+    :param definition: Definition of the metric.
+    :param grading_prompt: Grading criteria of the metric.
+    :param examples: (Optional) Examples of the metric.
+    :param version: (Optional) Version of the metric. Currently supported versions are: v1.
+    :param model: (Optional) Model uri of the metric.
+    :param variables: (Optional) Variables required to compute the metric.
+    :param parameters: (Optional) Parameters for the llm used to compute the metric.
+    :param aggregations: (Optional) The list of options to aggregate the scores. Currently supported
+        options are: min, max, mean, median, variance, p90.
+    :param greater_is_better: (Optional) Whether the metric is better when it is greater.
+
+    :return: A metric object.
+
+    .. test-code-block:: python
+        :caption: Example for creating a genai metric
+
+        from mlflow.metrics.base import EvaluationExample, make_genai_metric
+
+        example = EvaluationExample(
+            input="What is MLflow?",
+            output=(
+                "MLflow is an open-source platform for managing machine "
+                "learning workflows, including experiment tracking, model packaging, "
+                "versioning, and deployment, simplifying the ML lifecycle."
+            ),
+            score=4,
+            justification=(
+                "The definition effectively explains what MLflow is "
+                "its purpose, and its developer. It could be more concise for a 5-score.",
+            ),
+            variables={
+                "ground_truth": (
+                    "MLflow is an open-source platform for managing "
+                    "the end-to-end machine learning (ML) lifecycle. It was developed by "
+                    "Databricks, a company that specializes in big data and machine learning "
+                    "solutions. MLflow is designed to address the challenges that data "
+                    "scientists and machine learning engineers face when developing, training, "
+                    "and deploying machine learning models."
+                )
+            },
+        )
```

Comment:
@sunishsheth2009 was it possible to use triple quoted string here (and for other similar cases)? i think it would help a lot with readibility

Reply:
It wasn't unfortunately. Thats is why updated the code using @harupy technique. :)

Reply:
Ah i see now. sgtm